yt/data_objects/selection_data_containers.py

```diff
@@ -1163,12 +1163,16 @@ def _part_ind_KDTree(self, ptype):
             ],
             axis=1,
         ).value
+
+        mask = np.zeros(ppos.shape[0], dtype=bool)
+
         levels = self[("index", "grid_level")].astype("int32").value
+        if levels.size == 0:
```

Comment:
So this should trigger if there are no cells, right?  But `mask` here is the particle position?  So won't it give a differently shaped array in this case?  Or is the particle size the correct array shape?

Reply:
@cphyc likely knows better than me, but I think we only get the levels to get the grid cells on that level. We then use the kdtree to check the distances of those cells to the particle and mask out the ones that are too far away. If the levels array is empty, then we don't want to go into that loop at all.

Reply:
Yup, if there are no levels then we cannot loop over them and the code will likely fail.